libs/@blockprotocol/core/src/service-handler.ts

```diff
@@ -48,27 +63,56 @@ export abstract class ServiceHandler {
     serviceName,
     sourceType,
   }: {
-    element: HTMLElement;
+    element?: HTMLElement | null;
     serviceName: string;
     sourceType: "block" | "embedder";
   }) {
-    this.element = element ?? null;
     this.serviceName = serviceName;
     this.sourceType = sourceType;
 
+    if (element) {
+      this.registerService(element);
+    }
+  }
+
+  initialize(element: HTMLElement) {
+    if (!this.element) {
+      this.registerService(element);
+    }
```

Comment:
It is possible that this is called with an element different to that provided in the constructor, which we might want to at least throw an error on here.